tests/jballerina-unit-test/src/test/resources/test-src/runtime/api/identifier_utils/main.bal

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/jballerina.java;
+
+public function main() {
+    testIdentifierDecoding();
+    testEscapeSpecialCharacters();
+    testUnescapeSpecialCharacters();
+}
+
+function testIdentifierDecoding() {
+    test:assertEquals(decodeIdentifier("üňĩćőđę_ƈȏɳʂʈ_IL"), "üňĩćőđę_ƈȏɳʂʈ_IL");
+    test:assertEquals(decodeIdentifier("const_IL_123"), "const_IL_123");
+    test:assertEquals(decodeIdentifier(" $0047$0058@$0091`{~_IL"), " /:@[`{~_IL");
+}
+
+function testEscapeSpecialCharacters() {
+    test:assertEquals(escapeSpecialCharacters("'version"), "\\'version");
+    test:assertEquals(escapeSpecialCharacters(" ' $0047$0058@$0091`{~_IL"), "\\ \\'\\ \\$0047\\$0058\\@\\$0091\\`\\{\\~_IL");
+    test:assertEquals(escapeSpecialCharacters("üňĩćőđę_ƈȏɳʂʈ_IL"), "üňĩćőđę_ƈȏɳʂʈ_IL");
+    test:assertEquals(escapeSpecialCharacters("'ü?ňĩ$ć%őđ&ę_ƈ-ȏɳ+ʂʈ_=L#"), "\\'ü\\?ňĩ\\$ć\\%őđ\\&ę_ƈ\\-ȏɳ\\+ʂʈ_\\=L\\#");
+    test:assertEquals(escapeSpecialCharacters("\u{03C0}A9őđ&ę\u{00A9}\u{00AE}\u{2122}@3)^~`\u{03A3}\u{03A8}"), "πA9őđ\\&ę©®™\\@3\\)\\^\\~\\`ΣΨ");
+    test:assertEquals(escapeSpecialCharacters("\u{1f479}\u{1F47A}\u{1f47B}\u{1F63A}\u{1F408}\u{1F981}\u{1F600}"), "👹👺👻😺🐈🦁😀");
+}
+
+function testUnescapeSpecialCharacters() {
+    test:assertEquals(unescapeSpecialCharacters("\\'version"), "'version");
+    test:assertEquals(unescapeSpecialCharacters("\\ \\'\\ \\$0047\\$0058\\@\\$0091\\`\\{\\~_IL"), " ' $0047$0058@$0091`{~_IL");
+    test:assertEquals(unescapeSpecialCharacters("\\'ü\\?ňĩ\\$ć\\%őđ\\&ę_ƈ\\-ȏɳ\\+ʂʈ_\\=L\\#"), "'ü?ňĩ$ć%őđ&ę_ƈ-ȏɳ+ʂʈ_=L#");
+    test:assertEquals(unescapeSpecialCharacters("πA9őđ\\&ę©®™\\@3\\)\\^\\~\\`ΣΨ"), "πA9őđ&ę©®™@3)^~`ΣΨ");
+    test:assertEquals(unescapeSpecialCharacters("πA9őđ\\&ę©®™\\@3\\)\\^\\~\\`ΣΨ"), "\u{03C0}A9őđ&ę\u{00A9}\u{00AE}\u{2122}@3)^~`\u{03A3}\u{03A8}");
+    test:assertEquals(unescapeSpecialCharacters("👹👺👻😺🐈🦁😀\u{1f920}👐"), "\u{1f479}\u{1F47A}\u{1f47B}\u{1F63A}\u{1F408}🦁😀🤠\u{1F450}");
```

Comment:
Shall we check why `🦁😀` characters didn't get unescaped to `\u{1F981}\u{1F600}` as the other characters? It could be a bug. 
We can create an issue and fix it later.

Reply:
Assertion is succeed we have either `\u{1F981}\u{1F600}` or 🦁😀. Is it a wrong behavior?

Reply:
The above case is not due to a bug. Actually I missed to include some test cases with escape chars. Added some tests with https://github.com/ballerina-platform/ballerina-lang/pull/38318/commits/73efab5391b5670ed6b03b31040c7042599f9e0b.